.github/workflows/search-e2e.yml

```
@@ -0,0 +1,68 @@
+name: Enterprise Search e2e tests
+env:
+  NODE_VERSION: "16"
+  NODE_CACHE: "${{ github.workspace }}/node_modules_cache"
+
+on:
+  push:
+    branches:
+      - develop
+      - production
+      - staging
+  pull_request:
+    branches:
+      - develop
+      - production
+      - staging
+
+jobs:
+  cypress_local:
+    name: Cypress - Local
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v3
+      with:
+        submodules: recursive
+
+    - name: Prepare npm cache
+      uses: actions/cache@v2
+      with:
+        path: ${{ env.NODE_CACHE }}
+        key: npm-${{ env.NODE_VERSION }}-${{ hashFiles('**/package-lock.json') }}
+        restore-keys: |
+          npm-${{ env.NODE_VERSION }}-
+    - name: "Install node v${{ env.NODE_VERSION }}"
+      uses: actions/setup-node@v3
+      with:
+        node-version: ${{ env.NODE_VERSION }}
```

```suggestion
    - name: "Install node v${{ env.NODE_VERSION }}"
      uses: actions/setup-node@v3
      with:
        node-version: ${{ env.NODE_VERSION }}
        cache: npm
```

`actions/setup-node` now knows how to [cache dependencies](https://github.com/actions/setup-node#caching-global-packages-data).

PS: ${{ hashFiles('**/package-lock.json') }} is not good because we have unrelated lock files, like in Search DevTools.